src.ts/wallet/utxohdwallet.ts

```diff
@@ -374,7 +370,14 @@ export class UTXOHDWallet extends BaseWallet {
         if (!input.pub_key) throw new Error('Missing public key for input');
         const pubKey = input.pub_key;
         const address = this.getAddressFromPubKey(hexlify(pubKey));
-        const privKey = this.utxoAddresses.find(utxoAddr => utxoAddr.address === address)?.privKey;
+        // get shard from address
+        const shard = getShardForAddress(address);
+        if (!shard) throw new Error(`Invalid address: ${address}`);
```

Comment:
Change error message to `Invalid shard location for address: ${address}` for clarity